src/languages/rust.ts

```diff
@@ -0,0 +1,109 @@
+import {
+    createPatternMatchers,
+    argumentMatcher,
+    leadingMatcher,
+    conditionMatcher,
+    trailingMatcher,
+    childAtIndexMatcher,
+    cascadingMatcher,
+    patternMatcher,
+  } from "../util/nodeMatchers";
+  import {
+    NodeMatcherAlternative,
+    ScopeType,
+    SelectionWithEditor,
+  } from "../typings/Types";
+  import { getNodeRange } from "../util/nodeSelectors";
+  import { SyntaxNode } from "web-tree-sitter";
+  
+  // Generated by the following command:
+  // `curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-rust/master/src/node-types.json | jq '[.[] | select(.type == "_declaration_statement" or .type == "_expression") | .subtypes[].type]'`
+  const STATEMENT_TYPES = [
```

Comment:
I think _declaration_statement will be too narrow, it wouldn't include if statements for example

Reply:
Got it.  Looks like the Rust parser is like Ruby, where the `expression_statement` node doesn't appear in the parse tree.  The problem is that we need to distinguish between eg an `if` statement as a statement, vs `if` statement as an expression.   For example, when you say "take state", you'll capture the if statement here:

```rust
let foo = if bar { "hello" } else { "there" }
```

which you don't want, because that `if` statement is actually functioning as an expression.  In Ruby, we resorted to inspecting the parent node to see if the expression is at statement level vs a nested expression.  See https://github.com/cursorless-dev/cursorless-vscode/blob/993fbc8d14846414dbfc374c6fa56d3098f03465/src/languages/ruby.ts#L157-L166.  Should be able to do something similar here

Reply:
ahhhh I see what you are saying now